File: Tutoriels/CSharp/POO/src/manager/GProcess.cs
//===============================================
using System;
using System.Collections.Generic;
//===============================================
public abstract class GProcess {
    //===============================================
    public static GProcess Instance() {
        string lProcess = GConfig.Instance().getData("PROCESS");
        if(lProcess == "HELLO") return GProcessHello.Instance();
        return GProcessHello.Instance();
    }    
    //===============================================
    public virtual void run(string[] args) {
        GPrint.Instance().print("Bonjour tout le monde _DEFAULT_");
    }
    //===============================================
}
//===============================================

File: Tutoriels/Python/POO/src/manager/GProcess.py
#================================================
from GConfig import GConfig
#================================================
class GProcess:
    #================================================
    @staticmethod
    def Instance():
        lProcess = GConfig.Instance().getData("PROCESS")
        if lProcess == "HELLO": return GProcessHello.Instance()
        return GProcessHello.Instance()
    #================================================
    def run(self): pass
#================================================
from GProcessHello import GProcessHello
#================================================
